### ISSUE TITLE
Sound??

### ISSUE THREAD

**Issue by bionichuman** [phoboslab/jsmpeg-vnc#62]
Does sound work yet?? Trying to watch a video using this but sound does not seem to come through. Does sound work?

**Comment by eipeipkrop7992** [comment]
**Jsmpeg-vnc**  has not a JavaScript sound decoder implemented. I going to implement the js sound decoder and do a pull request. 

I needed a js browser video and sound decoder some year ago but I didn't found out.

**Comment by bionichuman** [comment]
Cool, please let me know when you do. I am trying to just unblock Netflix through a web proxy or this. If you have any ideas for that please let me know too.

**Comment by eipeipkrop7992** [comment]
What do you mean unblock Netflix with proxy?

**Comment by bionichuman** [comment]
@eipeipkrop7992 Access netflix via a web proxy.

**Comment by eipeipkrop7992** [comment]
Yeap I saw that. That is a MP2 decoder. Actually only in Jsmpeg-vnc repo it isn't not implemented. I will need to code sound grabber and encode that. It's not difficult. All this things is on my to-do list. 

So, ideas or support you can add me on Discord @eipeipkrop7992

**Comment by BenNottelling** [comment]
Hey @eipeipkrop7992 any progress at implementing audio?

**Comment by eipeipkrop7992** [comment]
Hi.

I'm at uni. I'm finishing my final thesis, so until January i will be busy.

**Comment by kitsonGit** [comment]
May I know if there are any updates for the sound grabber? 
I think your project is very interesting.

**Comment by gegenoketaru** [comment]
Hey,

Grab sound is not difficult. You can grab the whole desktop sound via drive
or grab it hooking the process buffer. Use a few FFmpeg parameters to
encode it and  after decode using a js decoder. Then merge into jsmpeg.
I'll try to find out a free time to implement this project.

Em qua, 10 de jul de 2019 às 16:40, kitsonGit <notifications@github.com>
escreveu:

**Comment by kitsonGit** [comment]
Hey,

It is found that my FFmpeg can only find recorder devices instead of playback devices. 
Then I use a VB virtual cable to redirect the sound to a virtual recorder device. The sound can be captured.

after following the instruction of "https://github.com/phoboslab/jsmpeg" 
I can stream the desktop with sound now.

I do not have much knowledge about c program. Hope it can give you some ideas for future improvement. 

Thanks

**Comment by thejammie9** [comment]
> Hey,
> 
> It is found that my FFmpeg can only find recorder devices instead of playback devices.
> Then I use a VB virtual cable to redirect the sound to a virtual recorder device. The sound can be captured.
> 
> after following the instruction of "https://github.com/phoboslab/jsmpeg"
> I can stream the desktop with sound now.
> 
> I do not have much knowledge about c program. Hope it can give you some ideas for future improvement.
> 
> Thanks

Would you be able to list how you managed to do it as i was reading what you read and it makes zero sense to me so i hint into the right direction would be awesome 

Thanks!
